# Questao 1

In [1]:
import pandas as pd
import sqlite3

In [2]:

# 1. Carregar o dataset fornecido
df_vendas = pd.read_csv('datasets/vendas_2023_2024.csv')

# 2. Criar uma conexão com um banco de dados SQLite em memória
conn = sqlite3.connect(':memory:')

# 3. Salvar o DataFrame como uma tabela SQL chamada 'vendas'
df_vendas.to_sql('vendas', conn, index=False, if_exists='replace')

print("Tabela 'vendas' estruturada com sucesso para consultas SQL.")

Tabela 'vendas' estruturada com sucesso para consultas SQL.


### SQL e validação (Questão 1.1 & 1.2)
Query para coletar informaçòes de datas (mínimo e máximo), e valroes (mínimo, máximo e média)

In [3]:
query = """
SELECT 
    COUNT(*) AS total_linhas,
    (SELECT COUNT(*) FROM pragma_table_info('vendas')) AS total_colunas,
    MIN(sale_date) AS data_minima,
    MAX(sale_date) AS data_maxima,
    MIN(total) AS valor_minimo,
    MAX(total) AS valor_maximo,
    AVG(total) AS valor_medio
FROM vendas;
"""

# Executar a query e exibir o resultado
resultado = pd.read_sql(query, conn)
resultado

,total_linhas,total_colunas,data_minima,data_maxima,valor_minimo,valor_maximo,valor_medio
0,9895,8,01-01-2023,31-12-2024,294.5,2222973.0,263797.828267


In [4]:
df_vendas.describe()

,Unnamed: 0,id,id_client,id_product,qtd,total
count,9895.000000,9895.000000,9895.000000,9895.000000,9895.000000,9.895000e+03
mean,4947.000000,5000.755533,24.874583,75.255786,8.015260,2.637978e+05
std,2856.584791,2887.000000,14.177715,43.533397,4.301723,3.900072e+05
min,0.000000,0.000000,1.000000,1.000000,1.000000,2.945000e+02
25%,2473.500000,2501.500000,13.000000,37.000000,4.000000,2.313820e+04
50%,4947.000000,4999.000000,25.000000,74.000000,8.000000,8.222500e+04
75%,7420.500000,7505.500000,37.000000,114.000000,12.000000,3.390945e+05
max,9894.000000,9999.000000,49.000000,150.000000,15.000000,2.222973e+06


In [5]:
df_vendas.head(10)

,Unnamed: 0,id,id_client,id_product,qtd,total,sale_date,sale_date_dt
0,0,0,42,105,11,3405.00,2023-09-10,2023-09-10
1,1,1,3,136,9,16873.90,15-09-2024,2024-09-15
2,2,2,25,139,7,9475.30,2024-08-13,2024-08-13
3,3,4,20,23,5,55893.00,2023-02-03,2023-02-03
4,4,5,8,57,4,451403.90,2024-02-12,2024-02-12
5,5,6,36,52,3,39056.40,2023-09-26,2023-09-26
6,6,8,27,25,3,34560.05,2024-02-28,2024-02-28
7,7,9,37,26,7,114932.90,07-11-2023,2023-07-11
8,8,10,31,143,3,12643.55,2024-08-25,2024-08-25
9,9,11,39,128,5,23254.00,2023-05-07,2023-05-07


In [6]:
conn.close()

### Interpretação (Questão 1.3)


***Comente sobre***

Outliers em "total: O código mostra valores máximos muito altos (milhões). Embora a LH Nautical venda motores caros (categoria Propulsão), é recomendado que esses valores sejam validados contra o catálogo de preços para garantir que não são erros de digitação.

Inconsistência de Datas: Ao observar os dados, nota-se que existem formatos mistos como YYYY-MM-DD (ex: 2023-09-10) e DD-MM-YYYY (ex: 15-09-2024). Isso significa que, sem tratamento, o computador não consegue ordenar as datas corretamente, sendo necessário escolher um formato para padroinzar as datas.

Qualidade e Confiabilidade: O dataset apresenta lacunas (IDs de venda pulados por exemplo). Portanto, o diagnóstico é que os dados não são confiáveis para decisão imediata e exigem uma etapa de Engenharia de Dados (limpeza) antes de qualquer aplicação de IA para prever vendas. Outra prova da inconsistência dos id de vendas se dá no fato de os ids irem de 0 a 9899, enquanto que so temos 9895 linhas.


